omod/src/main/java/org/openmrs/module/htmlformentryui/page/controller/htmlform/ViewEncounterWithHtmlFormPageController.java

```diff
@@ -65,5 +83,12 @@ public void get(@RequestParam("encounter") Encounter encounter,
     private String fixCase(String word) {
         return Character.toUpperCase(word.charAt(0)) + word.substring(1).toLowerCase();
     }
+    
+     @ExceptionHandler(value= NullPointerException.class)
+     public String HandleNullPointerException(Exception e) {
+    	  log.warn("encounter.form is not an HTML Form");
```

Comment:
doesn't this catch any Null Pointer Exception?  How do we know that the "encounter.form" NPE was the one that caused it?

Reply:
Ideally wanted this should not be actually nullpointer exception but rather any exception, have changed it to illegalArgumentException, this method would handle any excetion that would occur when any method is called